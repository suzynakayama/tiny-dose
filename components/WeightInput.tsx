import { StyleSheet, Text, TextInput, View } from 'react-native'
import React from 'react'
import { weightOptions } from '@/Types/medicineCalculator.contants'
import { SwitchType } from '@/Types/medicineCalculator.types'
import BasicSwitchSelector from './BasicSwitchSelector'

// This component allows the user to input the child's weight and switch between kg and lb units.

interface WeightInputProps {
  weight: string
  setWeight: (weight: string) => void
  changeSwitch: (value: string, type: SwitchType) => void
}

export default function WeightInput({ weight, setWeight, changeSwitch }: WeightInputProps) {
  return (
    <>
      <View style={styles.inputContainer}>
        <Text>Child weight:</Text>
        <TextInput
          style={styles.input}
          placeholder='Enter weight...'
          placeholderTextColor='#aaa'
          value={weight}
          onChangeText={newWeight => setWeight(newWeight)}
          keyboardType='numeric'
        />
      </View>
      <BasicSwitchSelector
        accessibilityLabel='weight-selector'
        testID='weight-selector'
        options={weightOptions}
        changeSwitch={changeSwitch}
        switchType='weight'
        selectorStyles={styles.weightSelector}
      />
    </>
  )
}

const styles = StyleSheet.create({
  input: {
    borderBottomWidth: 1,
    borderColor: '#ccc',
    fontSize: 16,
    marginLeft: 5,
    padding: 5,
  },
  inputContainer: {
    alignItems: 'center',
    flexDirection: 'row',
  },
  weightSelector: {
    padding: 3,
    width: 100,
  },
})
