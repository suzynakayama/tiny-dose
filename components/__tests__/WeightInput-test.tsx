import React from 'react'
import { render, screen } from '@testing-library/react-native'
import WeightInput from '../WeightInput'

test('renders correctly', () => {
  const weight = '20'
  const setWeight = jest.fn()
  const changeSwitch = jest.fn()

  const WeightInputComponent = render(
    <WeightInput weight={weight} setWeight={setWeight} changeSwitch={changeSwitch} />
  )

  const input = screen.getByPlaceholderText('Enter weight...')
  expect(input).toBeOnTheScreen()
  expect(input).toHaveProp('keyboardType', 'numeric')
  expect(input).toHaveProp(
    'style',
    expect.objectContaining({
      borderBottomWidth: 1,
      borderColor: '#ccc',
      fontSize: 16,
      marginLeft: 5,
      padding: 5,
    })
  )

  // test the switch selector and options to be showing on the screen
  const switchSelector = screen.getByTestId('weight-selector')
  expect(switchSelector).toBeOnTheScreen()

  const options = ['kg', 'lb']
  options.forEach(option => {
    expect(screen.getByText(option)).toBeOnTheScreen()
  })

  // Snapshot testing
  const WeightInputJson = WeightInputComponent.toJSON()
  expect(WeightInputJson).toMatchSnapshot()
})
