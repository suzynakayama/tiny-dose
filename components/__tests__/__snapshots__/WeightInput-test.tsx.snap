// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
[
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text>
      Child weight:
    </Text>
    <TextInput
      keyboardType="numeric"
      onChangeText={[Function]}
      placeholder="Enter weight..."
      placeholderTextColor="#aaa"
      style={
        {
          "borderBottomWidth": 1,
          "borderColor": "#ccc",
          "fontSize": 16,
          "marginLeft": 5,
          "padding": 5,
        }
      }
      value="20"
    />
  </View>,
  <View
    accessibilityLabel="weight-selector"
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "padding": 3,
          "width": 100,
        },
      ]
    }
    testID="weight-selector"
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        {
          "flex": 1,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          {
            "backgroundColor": "#f6bdc0",
            "borderRadius": 50,
            "height": 40,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "borderColor": "#C9C9C9",
              "borderRadius": 50,
              "borderWidth": 0,
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            accessibilityLabel="switch-to-kg"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "color": "#fff",
                    "fontSize": 14,
                    "fontWeight": "normal",
                    "textAlign": "center",
                  },
                  {
                    "fontWeight": "bold",
                    "textShadowColor": "#031F72",
                    "textShadowOffset": {
                      "height": 1,
                      "width": 1,
                    },
                    "textShadowRadius": 2,
                  },
                ]
              }
            >
              kg
            </Text>
          </View>
          <View
            accessibilityLabel="switch-to-lb"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "color": "#333",
                    "fontSize": 14,
                    "fontWeight": "normal",
                    "textAlign": "center",
                  },
                  {},
                ]
              }
            >
              lb
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>,
]
`;
